### DISCUSSION TITLE
Gif displaying previous frames in discord

### DISCUSSION THREAD

**Discussion by KredSuldaan** [ImageMagick/ImageMagick#3615] [Help] [answered]
I'm trying to make a gif for a discord emote and in windows photos app it looks like it should, but when I send it in discord it has previous frames in the background. the command I'm using is: "magick.exe convert -delay 10 -loop 0 -dispose previous -scale 50%"
I don't see what I'm doing wrong. Anyone know what I need to do?

**Comment by fmw42** [comment, marked as answer]
try

`magick.exe -dispose none -delay 10 image1 image2 ... imageN -scale 50% -layers optimize animation.gif`

Note that in ImageMagick 7, use magick, not convert and not magick convert. So on Windows, only magick.exe.

Please always provide a full proper command line and tell us what exact version of ImageMagick you are using.

**Reply by KredSuldaan** [reply]
sorry. im using imageMagick 7.0.11. and I tried your solution and there is no change. the full command I'm using now is `magick.exe -dispose none -delay 10 c:\users\user\videos\gif\frame*.png -scale 50% -layers optimize animation.gif`

**Reply by fmw42** [reply]
Can you post a few frames for us to use to test the command?  You can zip a few and post the zip file. P.S. Please in the future post the full version 7.0.11.x. You can get it from `magick.exe -version`

**Reply by KredSuldaan** [reply]
Sorry, i'm new to this. its version 7.0.11-q16-hdri.
[Gif.zip](https://github.com/ImageMagick/ImageMagick/files/6408737/Gif.zip)
The gif is only 5 frames so that's all of them

**Reply by fmw42** [reply]
Use `magick.exe -version` to get the proper version number next time. It is OK now. Thanks for the zip file.

This is what I get on my Mac using:

`magick -background none -dispose previous -delay 10 *.png -coalesce -scale 50% -layers optimize animation.gif`

Is this not what you expect?

![animation](https://user-images.githubusercontent.com/13458218/116767992-64962380-a9e8-11eb-8189-a1582dccda35.gif)

Change `magick` to `magick.exe` for Windows.

**Reply by fmw42** [reply]
See my edit. I replaced -dispose none with -dispose previous in the above.

**Reply by KredSuldaan** [reply]
it looks like that in the windows photos app, but in discord it has the previous frames in the background.

**Reply by fmw42** [reply]
Then I suspect the issue is with your discord app. I believe the ImageMagick is making a correct gif animation.

**Reply by KredSuldaan** [reply]
alright. thanks for the help

**Reply by fmw42** [reply]
try -dispose background

`magick -background none -dispose background -delay 10 *.png -coalesce -scale 50% -layers optimize animation.gif`


![animation](https://user-images.githubusercontent.com/13458218/116768500-eb003480-a9eb-11eb-858b-b0e66f9ba422.gif)

**Reply by KredSuldaan** [reply]
no change
